playwright/tests/a11y.spec.js

```diff
@@ -1,66 +1,145 @@
 
-  let pageSegments
+    if (segmentNumber !== 0) {
+      pages = splitArray(pages, totalSegments)[segmentNumber - 1]
+    }
 
-  test.beforeAll(async () => {
-    pageSegments = await splitPagesIntoSegments(pages, 5)
-  })
+    console.log('number of pages in segment', pages.length)
+    console.log('browser name:', browserName)
+    console.log('viewport size:', viewportSize)
+
+    for (const pageUrl of pages) {
+      if (excludedPages.includes(pageUrl)) {
+        console.log('skipping scan of excluded page:', pageUrl)
+        continue
+      }
+
+      try {
+        console.log('testing page:', pageUrl)
+        await page.goto(pageUrl)
+
+        // @todo The shared "makeAxeBuilder" never reports errors for whatever reason so not using for now.
+        // const accessibilityScanResults = await makeAxeBuilder({ page }).analyze()
+
+        const accessibilityScanResults = await new AxeBuilder({ page })
+          .withTags(['section508', 'wcag2a', 'wcag2aa', 'wcag21a', 'wcag21aa'])
+          .exclude('iframe')
+          // Exclude header and footer since they are shared on every page.
+          .exclude('header')
+          .exclude('footer')
+          .analyze()
+
+        console.log('page violations:', accessibilityScanResults.violations)
+        if (accessibilityScanResults.violations.length > 0) {
+          scanResults.push({
+            url: pageUrl,
+            browserName,
+            viewportSize,
+            violations: accessibilityScanResults.violations,
+          })
+        }
+      } catch (error) {
+        let message = error
+        if (error instanceof Error) {
+          message = error.message
+        }
+
+        console.log('error:', message)
+        failedPages.push({
+          url: pageUrl,
+          browserName,
+          viewportSize,
+          error: message,
+        })
+      }
+    }
 
-  for (let i = 0; i < 5; i++) {
-    test(`the site should be accessible - Segment ${i + 1}`, async ({
-      page,
-      makeAxeBuilder,
-    }) => {
-      await runA11yTestsForPages(
-        pageSegments[i],
-        `Segment ${i + 1}`,
-        page,
-        makeAxeBuilder
+    if (scanResults.length > 0) {
+      // Remove root array from scanResults so we can merge results more cleanly.
+      const trimmedScanResults =
+        JSON.stringify(scanResults, null, 2).replace(/^\[|]$/g, '') +
+        // Add a trailing comma to the output so JSON is valid when merged.
+        ',\n'
+
+      fs.writeFileSync(`segment-${segmentNumber}.json`, trimmedScanResults)
+    }
+
+    if (failedPages.length > 0) {
+      const trimmedFailedPages =
+        JSON.stringify(failedPages, null, 2).replace(/^\[|]$/g, '') +
+        // Add a trailing comma to the output so JSON is valid when merged.
+        ',\n'
+
+      fs.writeFileSync(
+        `failed-pages-segment-${segmentNumber}.json`,
+        trimmedFailedPages
       )
-    })
-  }
+    }
+  })
 })
+
+function splitArray(array, numChunks) {
```

Comment:
i'm curious why this was needed versus re-using the `splitPagesIntoBatches` and `getPagesSlice` from the sitemapLocations utility

Reply:
I don't fully remember but I was looking to split an array into equal chunks and the `splitPagesIntoBatches` wasn't giving me as equal of chunks. But could of been some other issue I was facing...

Reply:
I deleted that function I made and used `splitPagesIntoBatches` instead.